packages/python-packages/api-stub-generator/apistub/nodes/_function_node.py

```diff
@@ -0,0 +1,218 @@
+            if params[argname].kind in [Parameter.VAR_KEYWORD, Parameter.KEYWORD_ONLY]:
+                arg.argname = KW_ARG_NAME
+            self.args.append(arg)
+
+        if sig.return_annotation:
+            self.return_type = get_qualified_name(sig.return_annotation)
+
+        # parse docstring
+        self._parse_docstring()
+        # parse type hints
+        self._parse_typehint()
+
+    def _parse_docstring(self):
+        # Parse docstring to get list of keyword args, type and default value for both positional and
+        # kw args and return type( if not already found in signature)
+        docstring = ""
+        if hasattr(self.obj, "__doc__"):
+            docstring = getattr(self.obj, "__doc__")
+        # Refer docstring at class if this is constructor and docstring is missing for __init__
+        if (
+            not docstring
+            and self.name == "__init__"
+            and hasattr(self.parent_node.obj, "__doc__")
+        ):
+            docstring = getattr(self.parent_node.obj, "__doc__")
+
+        if docstring:
+            #  Parse doc string to find missing types, kwargs and return type
+            parsed_docstring = DocstringParser(docstring)
+            parsed_docstring.parse()
+            # Set return type if not already set
+            if not self.return_type and parsed_docstring.ret_type:
+                logging.debug(
+                    "Setting return type from docstring for method {}".format(self.name)
+                )
+                self.return_type = parsed_docstring.ret_type
+
+            # Update arg type from docstring if available and if argtype is missing from signatrue parsing
+            arg_type_dict = dict(
+                [(x.argname, x.argtype) for x in parsed_docstring.pos_args]
+            )
+            for pos_arg in self.args:
+                if not pos_arg.argtype:
+                    pos_arg.argtype = arg_type_dict.get(
+                        pos_arg.argname, pos_arg.argtype
+                    )
+
+            # add keyword args
+            if parsed_docstring.kw_args:
+                # Add seperator to differentiate pos_arg and keyword args
+                self.args.append(ArgType("*"))
+                parsed_docstring.kw_args.sort(key=operator.attrgetter("argname"))
+                self.args.extend(parsed_docstring.kw_args)
+                # remove arg with name "**kwarg and add at the end"
+                kwargs = [x for x in self.args if x.argname == KW_ARG_NAME]
+                if kwargs:
+                    kw_arg = kwargs[0]
+                    self.args.remove(kw_arg)
+                    self.args.append(kw_arg)
+
+    def _parse_typehint(self):
+        # Parse type hint to get return type and types for positional args
+        typehint_parser = TypeHintParser(self.obj)
+        # Find return type from type hint if return type is not already set
+        if not self.return_type:
+            typehint_parser.find_return_type()
```

Comment:
Same comment as above re using type hint parsing to validate consistent param types with the docstring